<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MaemoSettingsWidget</class>
 <widget class="QWidget" name="MaemoSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>550</width>
    <height>350</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Maemo Device Configurations</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0" colspan="2">
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="configurationLabel">
       <property name="text">
        <string>Configuration:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="configurationComboBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="editable">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="2" rowspan="2">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="addConfigButton">
       <property name="text">
        <string>Add</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeConfigButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Remove</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="testConfigButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Test</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="deployKeyButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Deploy Key ...</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QFrame" name="frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QWidget" name="detailsWidget" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="deviceNameLabel">
           <property name="text">
            <string>Name</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="nameLineEdit"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="deviceTypeLabel">
           <property name="text">
            <string>Device type:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QWidget" name="deviceTypeButtonsWidget" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QRadioButton" name="deviceButton">
              <property name="text">
               <string>Remote device</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="simulatorButton">
              <property name="text">
               <string>Local simulator</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="authTypeLabel">
           <property name="text">
            <string>Authentication type:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QWidget" name="authTypeButtonsWidget" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QRadioButton" name="passwordButton">
              <property name="text">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="keyButton">
              <property name="text">
               <string>Key</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="hostNameLabel">
           <property name="text">
            <string>Host name:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="hostLineEdit">
           <property name="placeholderText">
            <string>IP or host name of the device</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="portsLabel">
           <property name="text">
            <string>Ports:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="sshPortLabel">
             <property name="text">
              <string>SSH:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sshPortSpinBox">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>65535</number>
             </property>
             <property name="value">
              <number>22</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="gdbServerLabel">
             <property name="text">
              <string>Gdb server:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="gdbServerPortSpinBox">
             <property name="maximum">
              <number>65535</number>
             </property>
             <property name="value">
              <number>10000</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="connectionTimeoutLabel">
           <property name="text">
            <string>Connection timeout:</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLineEdit" name="timeoutLineEdit">
           <property name="placeholderText">
            <string>Timeout value in milliseconds</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="userNameLabel">
           <property name="text">
            <string>User name:</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QLineEdit" name="userLineEdit"/>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="passwordLabel">
           <property name="text">
            <string>Password:</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QLineEdit" name="pwdLineEdit">
           <property name="echoMode">
            <enum>QLineEdit::Password</enum>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="keyLabel">
           <property name="text">
            <string>Private key file:</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="Utils::PathChooser" name="keyFileLineEdit" native="true"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1">
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
   <slots>
    <signal>editingFinished()</signal>
    <signal>browsingFinished()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>deviceButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>deviceTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>273</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hostLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>hostNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>163</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timeoutLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>timeoutEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>225</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>userLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>userNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>255</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pwdLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>passwordEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulatorButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>deviceTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>addConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>465</x>
     <y>27</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>deleteConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>passwordButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>authenticationTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>273</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>240</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyFileLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>keyFileEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyFileLineEdit</sender>
   <signal>browsingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>keyFileEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>testConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>testConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>102</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deployKeyButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>deployKey()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>523</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>458</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>authenticationTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>525</x>
     <y>408</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoSettingsWidget</receiver>
   <slot>configNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>330</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>configNameEditingFinished()</slot>
  <slot>deviceTypeChanged()</slot>
  <slot>hostNameEditingFinished()</slot>
  <slot>sshPortEditingFinished()</slot>
  <slot>timeoutEditingFinished()</slot>
  <slot>userNameEditingFinished()</slot>
  <slot>passwordEditingFinished()</slot>
  <slot>addConfig()</slot>
  <slot>selectionChanged()</slot>
  <slot>deleteConfig()</slot>
  <slot>authenticationTypeChanged()</slot>
  <slot>keyFileEditingFinished()</slot>
  <slot>testConfig()</slot>
  <slot>deployKey()</slot>
  <slot>gdbServerPortEditingFinished()</slot>
  <slot>currentConfigChanged(int)</slot>
 </slots>
</ui>
